src/boot/efi/boot.c

```diff
@@ -16,6 +16,11 @@
 #define EFI_OS_INDICATIONS_BOOT_TO_FW_UI 0x0000000000000001ULL
 #endif
 
+/* Some BIOSes scan the boot loader binary for the string "Microsoft" and
+ * deliver broken ACPI tables if it's not present.
+ */
+const char bios_confuser[] = "Microsoft";
```

Comment:
Is there any chance the compiler or linker may decide to optimize this out at some point in the future? Can we make it more "optimization-proof"?

Reply:
I think there isn't - this is linked into a .so where it's an exported symbol so it'll live on.

Then the final step in the build is objcopy to an efi binary - objcopy doesn't have the smarts to elide symbols.

However, it might also be possible for me to just objcopy a text file with the string "Microsoft" in it into the binary as the last step...  I could give that a try if you think it'll be safer.

Reply:
As discussed on slack, lets leave this as-is and add a test that checks for the string presence in the final binary, so we catch any regressions.